src/main/java/com/devcycle/sdk/server/local/bucketing/LocalBucketing.java

```diff
@@ -143,5 +155,77 @@ public BucketedUserConfig generateBucketedConfig(String token, String user) thro
+        int tokenAddress = newWasmString(token);
+        int userAddress = newWasmString(user);
+        int eventAddress = newWasmString(event);
+
+        Func queueEventPtr = linker.get(store, "", "queueEvent").get().func();
+        WasmFunctions.Consumer3<Integer, Integer, Integer> fn = WasmFunctions.consumer(store, queueEventPtr, I32, I32, I32);
+        fn.accept(tokenAddress, userAddress, eventAddress);
+    }
+
+    public void queueAggregateEvent(String token, String event, String variableVariationMap) {
+        int tokenAddress = newWasmString(token);
+        int eventAddress = newWasmString(event);
+        int variableVariationMapAddress = newWasmString(variableVariationMap);
+
+        Func queueAggregateEventPtr = linker.get(store, "", "queueAggregateEvent").get().func();
+        WasmFunctions.Consumer3<Integer, Integer, Integer> fn = WasmFunctions.consumer(store, queueAggregateEventPtr, I32, I32, I32);
+        fn.accept(tokenAddress, eventAddress, variableVariationMapAddress);
+    }
+
+    public EventPayload[] flushEventQueue(String token) throws JsonProcessingException {
+        int tokenAddress = newWasmString(token);
+
+        Func flushEventQueuePtr = linker.get(store, "", "flushEventQueue").get().func();
+        WasmFunctions.Function1<Integer, Integer> fn = WasmFunctions.func(
+                store, flushEventQueuePtr, I32, I32);
+
+        int resultAddress = fn.call(tokenAddress);
+        String flushPayloadsStr = readWasmString(resultAddress);
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+
+        SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ"); //2022-09-08T20:16:31.741Z
+        df.setTimeZone(TimeZone.getTimeZone("UTC"));
+        objectMapper.setDateFormat(df);
+
+        String test = "[{\"records\":[{\"user\":{\"user_id\":\"j_test\",\"deviceModel\":\"\",\"platform\":\"Java\",\"platformVersion\":\"11.0.16.1\",\"sdkType\":\"server\",\"sdkVersion\":\"1.1.0\"},\"events\":[{\"type\":\"customEvent\",\"target\":\"target\",\"customType\":\"test\",\"user_id\":\"j_test\",\"featureVars\":{\"62fbf6566f1ba302829f9e32\":\"62fbf6566f1ba302829f9e39\"}}]";
+
+        EventPayload[] payloads = objectMapper.readValue(flushPayloadsStr, EventPayload[].class);
+//        EventPayload[] payloads = objectMapper.readerFor(EventPayload[].class).readValue(test);
+
+        return payloads;
+    }
+
+    public void onPayloadFailure(String token, String payloadId, boolean retryable) {
+        int tokenAddress = newWasmString(token);
+        int payloadIdAddress = newWasmString(payloadId);
+
+        Func onPayloadFailurePtr = linker.get(store, "", "onPayloadFailure").get().func();
+        WasmFunctions.Consumer3<Integer, Integer, Integer> fn = WasmFunctions.consumer(store, onPayloadFailurePtr, I32, I32, I32);
+        fn.accept(tokenAddress, payloadIdAddress, retryable ? 1 : 0);
+    }
```

Comment:
So this function is going to be an issue - we have to do some more work because it has a non-standard process. 

All function parameters are pointers still - so I'm looking into the actual type id that we need for a uint (8/16/32/tbd) - which we'll need another memory copy function. I'd comment the body of this one out for now.

Reply:
oh weird, casting retryable to an int is working consistently (as in, the event queue is cleared when retryable is false and kept when it's true)

Reply:
Remember that you're giving a memory address to read from, so it's entirely  possible that the address of 0 and address of 1 is correct from your testing

Reply:
ah gotcha, okay i'll comment out for now